### ISSUE TITLE
Cannot apply a large-ish Grafana dashboard configmap

### ISSUE THREAD

**Issue by LukeChannings** [rancher/k3s#1809]
**Version:**
`k3s version v1.18.2+k3s1 (698e444a)`

**K3s arguments:**

`k3s server --disable traefik --disable servicelb --cluster-cidr 172.30.0.0/16 --service-cidr 172.31.0.0/16 --cluster-dns 172.31.0.10`

**Describe the bug**

I am creating a configmap containing a Grafana dashboard, as described [here](https://github.com/helm/charts/tree/master/stable/grafana#sidecar-for-dashboards).

The configmap's file size is 896K, this is a single dashboard.

**To Reproduce**
The configmap can be found [here](https://gist.github.com/LukeChannings/7952f071a9b941ba47081d9654778ac7)

**Expected behaviour**

The configmap applies successfully 

**Actual behaviour**

When I apply it I get `The ConfigMap "sample-grafana-dashboard" is invalid: metadata.annotations: Too long: must have at most 262144 bytes`

**Additional context / logs**

k3s is using the regular sqlite3 database

**Comment by brandond** [comment]
This is not a k3s issue. See: 
https://github.com/kubernetes/kubectl/issues/712
https://github.com/m-lab/prometheus-support/issues/218

tl;dr `kubectl apply` stores a copy of the applied configuration in an annotation so that it can calculate the difference and update the resource as necessary next time you do an apply. Unfortunately, with larger objects, the applied configuration can exceed the storage available to an annotation.

I believe the current suggested work-around is to split the Grafana configuration into multiple configmaps that don't exceed the maximum size. You can then mount all the configmaps in a directory, and have Grafana load them from there.

Long term you might consider an alternate workflow that doesn't involve storing large files as Kubernetes objects, since it's not really designed to accommodate that.

**Comment by LukeChannings** [comment]
I see,

I was using dashboards that are available on grafana.com and it turns out the chart has a script that automatically [import dashboards](https://github.com/helm/charts/tree/master/stable/grafana#import-dashboards), I just went for the ConfigMap option first because it's a bit more ergonomic.

If I was defining my own dashboard it probably wouldn't hit the size limit.

Thanks for your help!
